### DISCUSSION TITLE
Using non-custom OPUS encoder

### DISCUSSION THREAD

**Discussion by npostavs** [jamulussoftware/jamulus#1920] [General]
I tried switching to the non-custom OPUS encoder to see if it might result in better sound. But when I tried making the changes I got no sound coming through whatsoever. Anyone have any ideas and/or has made their own (hopefully more successful) attempt?

My code is here: https://github.com/jamulussoftware/jamulus/compare/master...npostavs:np.stdopus.v0

(related https://github.com/jamulussoftware/jamulus/discussions/1313#discussioncomment-611826_)

**Comment by dcorson-ticino-com** [comment]
The biggest problem is that the current protocol requires known packet sizes so use of variable length packets is not possible.  What encoder type did you change too?
I had once made an attempt at just transfering the PCM data without encoder, it worked at times, but as I had too many changes made because I didn't completely understand how the protocol works I quit and may someday start over.

**Reply by npostavs** [reply]
I kept the fixed length encoder setting:

```cpp
OpusEncoderStereo = opus_encoder_create ( SYSTEM_SAMPLE_RATE_HZ, 2, OPUS_APPLICATION_RESTRICTED_LOWDELAY, &iOpusError );
[...]
opus_encoder_ctl ( OpusEncoderStereo, OPUS_SET_VBR ( 0 ) );
```
